tools/timelock_proposal/mcm_with_timelock.go

```diff
@@ -0,0 +1,149 @@
+package timelock_proposal
+
+import (
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/smartcontractkit/ccip-owner-contracts/tools/errors"
+	owner "github.com/smartcontractkit/ccip-owner-contracts/tools/gethwrappers"
+	"github.com/smartcontractkit/ccip-owner-contracts/tools/mcms_proposal"
+)
+
+var ZERO_HASH = common.BytesToHash([]byte{})
+
+type MCMSWithTimelockChainMetadata struct {
+	mcms_proposal.ChainMetadata
+	TimelockAddress common.Address `json:"timelockAddress"`
+}
+
+type TimelockOperation string
+
+const (
+	Schedule TimelockOperation = "schedule"
+	Cancel   TimelockOperation = "cancel"
+	Bypass   TimelockOperation = "bypass"
+)
+
+type MCMSWithTimelockProposal struct {
+	mcms_proposal.Proposal
+
+	Operation TimelockOperation `json:"operation"` // Always 'schedule', 'cancel', or 'bypass'
+
+	// TODO: this should be configurable as a human-readable string
+	// i.e. 1d, 1w, 1m, 1y
+	MinDelay string `json:"minDelay"`
```

Comment:
why human readable instead of a `time.Duration`?

Reply:
Aren't we assuming that these proposals will be generated by a chainlink-deployments migration? I would assume that the input into the migration would be human readable. 

I would argue that since we're building a library, it's ok to assume users wrapper will be able to generate a `time.Duration`

Reply:
right but this proposal format will be written somewhere as sigs are aggregated, so this value will need to get written somewhere, what would that look like?

Reply:
Seems fine to leave it as a string, if we use the https://pkg.go.dev/time#ParseDuration function from the time library. 

In that case, would it make sense to use the errors directly from the library rather than wrapping them? Is there additional information that we add from wrapping it compared to https://cs.opensource.google/go/go/+/refs/tags/go1.23.0:src/time/format.go;l=1686?

Reply:
yeah probably don't need the custom error type here, will remove